inntekt/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/inntekt/MapInntekt.kt

```diff
@@ -4,13 +4,36 @@ import no.nav.helsearbeidsgiver.felles.Inntekt
 import no.nav.helsearbeidsgiver.felles.MottattHistoriskInntekt
 import no.nav.helsearbeidsgiver.inntekt.InntektskomponentResponse
 
-fun mapInntekt(response: InntektskomponentResponse): Inntekt {
-    val list = mutableListOf<MottattHistoriskInntekt>()
-    response.arbeidsInntektMaaned?.forEach { maaned ->
-        maaned.arbeidsInntektInformasjon?.inntektListe?.map { inntekt ->
-            list.add(MottattHistoriskInntekt(maaned.aarMaaned, inntekt.beloep ?: 0.0))
-        }
+fun mapInntekt(response: InntektskomponentResponse, orgnr: String): Inntekt {
+    if (orgnr.isNullOrEmpty()) {
+        return Inntekt(emptyList())
     }
-    val total: Double = list.sumOf { it.inntekt ?: 0.0 }
-    return Inntekt(total, list)
+    val list = response.arbeidsInntektMaaned?.map { it.aarMaaned to it.arbeidsInntektInformasjon?.inntektListe }
+        ?.toMap()
+        ?.filterKeyAndValueNotNull()
+        ?.flatMap { (maaned, inntekter) ->
+            inntekter.filter {
+                it.virksomhet?.identifikator.equals(orgnr)
```

Comment:
equals -> `==`

Reply:
Såvidt jeg husker så er det en forskjell mellom Java og Kotlin at førstnevnte krever `equals` for strengsammenligning, mens sistnevnte ikke gjør det.